src/lib/Libifl/pbsD_connect.c

```diff
@@ -254,101 +216,30 @@ get_hostsockaddr(char *host, struct sockaddr_in *sap)
 }
 
 /**
- * @brief
- *	Makes a PBS_BATCH_Connect request to 'server'.
+ * @brief	This function establishes the network connection to the choose server.
+ *
+ * @param[in]   server - The hostname of the pbs server to connect to.
+ * @param[in]   port - Port number of the pbs server to connect to.
+ * @param[in]   extend_data - a string to send as "extend" data
  *
- * @param[in]   server - the hostname of the pbs server to connect to.
- * @param[in]   extend_data - a string to send as "extend" data.
  *
  * @return int
- * @retval >= 0	index to the internal connection table representing the
- *		connection made.
+ * @retval >= 0	The physical server socket.
  * @retval -1	error encountered setting up the connection.
  */
-int
-__pbs_connect_extend(char *server, char *extend_data)
+
+static int
+tcp_connect(char *server, int server_port, char *extend_data)
 {
+	int i;
+	int sd;
+	int rc;
 	struct sockaddr_in server_addr;
 	struct sockaddr_in my_sockaddr;
-	int sock;
-	int i;
-	int f;
-	char  *altservers[2];
-	int    have_alt = 0;
 	struct batch_reply	*reply;
-	char server_name[PBS_MAXSERVERNAME+1];
-	unsigned int server_port;
 	char errbuf[LOG_BUF_SIZE] = {'\0'};
 
-#ifndef WIN32
-	char   pbsrc[_POSIX_PATH_MAX];
-	struct stat sb;
-	int    using_secondary = 0;
-#endif  /* not WIN32 */
-
-	/* initialize the thread context data, if not already initialized */
-	if (pbs_client_thread_init_thread_context() != 0)
-		return -1;
-
-	if (pbs_loadconf(0) == 0)
-		return -1;
-
-	/* get server host and port	*/
-
-	server = PBS_get_server(server, server_name, &server_port);
-	if (server == NULL) {
-		pbs_errno = PBSE_NOSERVER;
-		return -1;
-	}
-
-	if (pbs_conf.pbs_primary && pbs_conf.pbs_secondary) {
```

Comment:
This code actually handles failover code. Failover may not work without this. Are we going to handle it in another place by another method or so ?

Reply:
we have plans to deprecate the existing failover infrastructure, there will be an announcement/design discussion for the same soon.